lib/utils/httpRequestor.js

```diff
@@ -99,59 +95,71 @@ exports.create = function(requestorConfig) {
   };
 
-  var postFile = (options, callback) => {
-    var requestExtension = { body: getFileBody(options) };
-
-    return methodRequest(options, request.postAsync, 'POST', callback, requestExtension);
-  };
+  var postFile = (options, callback) => methodRequest(options, request.post, 'POST', callback, getFileBody(options));
 
   var deleteFunc = (options, callback)  =>
-    methodRequest(options, request.delAsync, 'DELETE', callback);
+    methodRequest(options, request.delete, 'DELETE', callback);
 
-  var put = (options, callback) => {
-    var requestExtension = { body: JSON.stringify(options.body) };
+  var put = (options, callback) => methodRequest(options, request.put, 'PUT', callback, options.body);
 
-    return methodRequest(options, request.putAsync, 'PUT', callback, requestExtension);
-  };
-
-  var methodRequest = (options, method, methodName, callback, requestExtension) => {
+  var methodRequest = (options, method, methodName, callback, body) => {
     var baseRequestOptions = {
       url: buildUrl(options),
       headers: buildHeaders(options),
-      qs: options.queryParameters,
-      encoding: options.encoding,
-      gzip: true
+      params: options.queryParameters,
+      responseEncoding: options.encoding
     };
-    var requestOptions = _.extend(baseRequestOptions, requestExtension);
+    var url = buildUrl(options);
+    var requestOptions = baseRequestOptions;
 
     var retryOptions = _.pick(options, 'maxRetryDurationMillis', 'calcRetryBackoff');
 
     logger.logRequest(methodName, requestOptions);
 
-    return makeRequestWithRetries(method, methodName, requestOptions, retryOptions, callback);
+    return makeRequestWithRetries(url, method, methodName, requestOptions, body, retryOptions, callback);
   };
 
-  var makeRequestWithRetries = (method, methodName, requestOptions, retryOptions, callback) => {
+  var makeRequestWithRetries = (url, method, methodName, requestOptions, body, retryOptions, callback) => {
     var effectiveRetryOptions = _.defaults(retryOptions, defaultRetryOptions);
 
     effectiveRetryOptions.endRetryTime = Date.now() + effectiveRetryOptions.maxRetryDurationMillis;
 
-    return retryHelper(method, methodName, requestOptions, effectiveRetryOptions, 0)
-      .tap(logger.logSuccessfulResponse)
-      .tapCatch(error => logger.logErrorResponse(methodName, requestOptions, error))
-      .get('content')
-      .catch(error => Promise.reject(_.omit(error, 'headers', 'body')))
-      .nodeify(callback);
+    return retryHelper(url, method, methodName, requestOptions, body, effectiveRetryOptions, 0)
+      .then((response) => {
+        logger.logSuccessfulResponse(response);
+
+        if (callback) {
+          callback(undefined, response.content);
+        }
+        
+        return response.content;
+      })
+      .catch((error) => {
+        logger.logErrorResponse(methodName, requestOptions, error);
+
+        if (callback) {
+          callback(error, undefined);
```

Comment:
I realize this is unchanged from your previous PR, but in looking at it again I'm wondering if the existing behavior is to strip off `headers` and `body` prior to passing the error to the callback? Is the `nodeify`'d callback passed the return value of `.catch` in Bluebird?

Reply:
Looking at this again I believe that this change preserves the initial callback behavior. The two parameters accepted are an `error` and a `success` object. In the case of failure there isn't going to be a success response from the promise chain. If the network request was the cause of the failure the body/headers should be funneled into the `error` object passed into the callback which would be the behavior I would expect from `nodeify`.